### PR TITLE
Extract exception message from chain for API errors

### PR DESCRIPTION
**Pull request by ludeeus**
## Proposed change

While working on #2099 I noticed that many operations would result in a blank error message returned to the CLI/API, this change extracts the message from the exception chain

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast supervisor tests`)
- [x] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

File: supervisor/addons/__init__.py [hunk 1]
```diff
@@ -169,7 +169,7 @@ async def install(self, slug: str) -> None:
             await addon.instance.install(store.version, store.image)
         except DockerError as err:
             self.data.uninstall(addon)
-            raise AddonsError() from err
+            raise AddonsError(err) from err
```
Comment (pvizeli) [inline supervisor/addons/__init__.py:172]: Do we need raise from err if we own the error? Important that our exception chain checker work with the new structure as well

Comment (ludeeus) [reply supervisor/addons/__init__.py:172]: Not sure, it should, I'll add a test for it

**Comment by pvizeli** [comment]
Your PR for handle error on API like for port mapping is not needed with https://github.com/home-assistant/supervisor/issues/1946

I approved it anyway, because it's don't change to match. Now we start to change the hole error handling to parse error message with regex on API return. It feel a bit like an hack.

I think instead to move forward with this, we should implement #1946 and allow to set usable error message where they are see (Like on run/start function on docker)

File: supervisor/addons/__init__.py (hunk 1 above)
Comment (ludeeus) [reply supervisor/addons/__init__.py:172]: Actually, check the current tests for `check_exception_chain` it seems to cover all cases already?

**Review by pvizeli — APPROVED** [review]

**Pull request merged**